ddtrace/mocktracer/mocktracer.go

```diff
@@ -40,13 +41,33 @@ type Tracer interface {
 // to activate the mock tracer. When your test runs, use the returned
 // interface to query the tracer's state.
 func Start() Tracer {
+	return StartWithOptions()
```

Comment:
Please revert all the changes to the mocktracer. They don't add anything. The tests you wrote in `contrib/net/http` do not guarantee at all that the service name is or is not inherited, they simply test the mock tracer's behavior.